homeassistant/components/homekit/type_cameras.py

```diff
@@ -190,18 +196,28 @@ def __init__(self, hass, driver, name, entity_id, aid, config):
             category=CATEGORY_CAMERA,
             options=options,
         )
+
         self._char_motion_detected = None
         self.linked_motion_sensor = self.config.get(CONF_LINKED_MOTION_SENSOR)
-        if not self.linked_motion_sensor:
-            return
-        state = self.hass.states.get(self.linked_motion_sensor)
-        if not state:
-            return
-        serv_motion = self.add_preload_service(SERV_MOTION_SENSOR)
-        self._char_motion_detected = serv_motion.configure_char(
-            CHAR_MOTION_DETECTED, value=False
-        )
-        self._async_update_motion_state(state)
+        if self.linked_motion_sensor:
+            state = self.hass.states.get(self.linked_motion_sensor)
+            if state:
+                serv_motion = self.add_preload_service(SERV_MOTION_SENSOR)
+                self._char_motion_detected = serv_motion.configure_char(
+                    CHAR_MOTION_DETECTED, value=False
+                )
+                self._async_update_motion_state(state)
+
+        self._char_doorbell_detected = None
+        self.linked_doorbell_sensor = self.config.get(CONF_LINKED_DOORBELL_SENSOR)
+        if self.linked_doorbell_sensor:
+            state = self.hass.states.get(self.linked_doorbell_sensor)
+            if state:
+                serv_doorbell = self.add_preload_service(SERV_DOORBELL)
```

Comment:
Do we need to make the doorbell the primary service for homekit to see this as a video doorbell ?

Reply:
I do not believe so. I did not read HAP Spec, but looked at most homebridge implementations. This seems to be the bare minimum required. The only visible change this makes to the camera in Home.app is in the notification settings for the accessory. It adds a toggle for Doorbell Notifications, which sends a push notification saying this accessory rang.